frontend/app/routes/create.$type.tsx

```diff
@@ -115,25 +159,82 @@ export default function Create() {
       )
       setVersionOptions(filteredOptions)
       setSelectedVersion('default')
-      setConfirmModalOpen(false)
-
-      const formData = new FormData()
-      formData.append('intent', 'remove')
-      formData.append('version', 'default')
-      formData.append('fullconfig', JSON.stringify(rest))
-      const defaultSet = rest.default as unknown as Record<string, string>
-      Object.keys(defaultSet).map((key) => {
-        formData.append(key, defaultSet[key])
-      })
+      setModalOpen(undefined)
+
+      const formData = getFormData(rest, 'remove', 'default')
       submitForm(formData, { method: 'post' })
     }
   }
 
+  const onConfirmOwnership = () => {
+    if (response?.grantRequired) {
+      window.location.href = response.grantRequired
+    }
+  }
+
+  const onResubmit = () => {
+    setModalOpen(undefined)
+    const formData = getFormData(fullConfig || {}, lastAction, selectedVersion)
+    submitForm(formData, { method: 'post', action: location.pathname })
+  }
+
+  const getConfirmModalContent = (type: string) => {
+    let title = '',
+      description = '',
+      onConfirm = () => {}
+    const onClose = () => setModalOpen(undefined)
+
+    switch (type) {
+      case 'confirm':
+        title = `Are you sure you want to remove ${selectedVersion}?`
+        onConfirm = onConfirmRemove
+        break
+      case 'wallet-ownership':
+        title = `Please confirm you are owner of 
+                <span class="flex w-full justify-center text-center">
+                  ${walletAddress.id}
+                </span>`
```

Comment:
Can we pass JSX nodes instead of `dangerouslySetInnerHTML`? `title` would then be `ReactNode` or `string | ReactElement`.